src/Akka.Streams.Kafka/Settings/CommitterSettings.cs

```diff
@@ -1,25 +1,65 @@
 using System;
 using Akka.Actor;
 using Akka.Configuration;
+using Akka.Streams.Kafka.Messages;
 
 namespace Akka.Streams.Kafka.Settings
 {
+    /// <summary>
+    /// Determines when to commit offsets
+    /// </summary>
+    public interface ICommitWhen;
+
+    public static class CommitWhen
```

Comment:
`ICommitWhen` strategy - specifies when we want to commit:

1. `OffsetFirstObserved` - commit observations as we see them
2. `NextOffsetObserved` - when commit A arrives, don't commit it until B arrives.

The reason for this is that `NextOffsetObserved` is a little bit tighter from an "at-least-once-delivery" standpoint, but it does run the higher risk of generating a duplicate - whereas `OffsetFirstObserved` commits more aggressively.